warehouse/query-core/src/main/java/datawave/query/jexl/visitors/JexlStringBuildingVisitor.java

```diff
@@ -185,24 +383,30 @@ public Object visit(ASTOrNode node, Object data) {
     
     public Object visit(ASTAndNode node, Object data) {
         StringBuilder sb = (StringBuilder) data;
-        
+        boolean needNewLines = needNewLines(node);
         int numChildren = node.jjtGetNumChildren();
-        
         JexlNode parent = node.jjtGetParent();
         boolean wrapIt = false;
+        
         if (!(parent instanceof ASTReferenceExpression || parent instanceof ASTJexlScript || parent instanceof ASTAndNode || numChildren == 0)) {
             wrapIt = true;
             sb.append("(");
         }
         
         Collection<String> childStrings = (sortDedupeChildren) ? new TreeSet<>() : new ArrayList<>(numChildren);
+        Collection<String> childStringsFormatted = (sortDedupeChildren) ? new TreeSet<>() : new ArrayList<>(numChildren);
         StringBuilder childSB = new StringBuilder();
         for (int i = 0; i < numChildren; i++) {
             node.jjtGetChild(i).jjtAccept(this, childSB);
             childStrings.add(childSB.toString());
             childSB.setLength(0);
         }
-        sb.append(String.join(" && ", childStrings));
+        // If needNewLines is false, we should remove the new lines added to the child strings
```

Comment:
I will have to say that this one scares me a little.  Since we have mergde the whole formatting thing into this one class (which is not a bad idea), we have to be very careful that the unformatted query path does not change the query.  If one of the nodes contains a term that actually has a newline in it, then I think that needs to be preserved.  So if buildMultipleLines is false, then we should never be adding any new lines in the first place.

Reply:
Good catch! I overlooked this. I will fix this.